app/controllers/payment-links/post-web-address.controller.js

```diff
@@ -12,26 +12,36 @@ const makeNiceURL = string => {
   return slugify(removeIndefiniteArticles(string))
 }
 
-module.exports = (req, res) => {
-  const pageData = lodash.get(req, 'session.pageData.createPaymentLink', {})
-  let updatedPageData = lodash.cloneDeep(pageData)
-
-  if (updatedPageData.productNamePath === '') {
-    req.flash('genericError', `<h2>There was a problem with the details you gave for:</h2><ul class="govuk-list govuk-error-summary__list"><li><a href="#payment-name-path">Please change the website address</a></li></ul>`)
-    return res.redirect(paths.paymentLinks.webAddress)
+module.exports = async function postWebAddress (req, res, next) {
+  const paymentLinkData = lodash.get(req, 'session.pageData.createPaymentLink')
+  if (!paymentLinkData) {
+    next(new Error('Payment link data not found in session cookie'))
   }
 
-  updatedPageData.productNamePath = makeNiceURL(req.body['payment-name-path'])
-  lodash.set(req, 'session.pageData.createPaymentLink', updatedPageData)
-
-  productsClient.product.getByProductPath(updatedPageData.serviceNamePath, updatedPageData.productNamePath)
-    .then(product => {
-    // if product exists we need to alert the user they must use a different URL
-      req.flash('genericError', `<h2>There was a problem with the details you gave for:</h2><ul class="govuk-list govuk-error-summary__list"><li><a href="#payment-name-path">Website address</a></li></ul>`)
-      return res.redirect(paths.paymentLinks.webAddress)
-    })
-    .catch((err) => { // eslint-disable-line handle-callback-err
-    // if it errors then it means no product was found and that’s good
+  const path = req.body['payment-name-path']
```

Comment:
Very minor but path in the context of a top level HTTP controller was ambitious here, something like `paymentLinkURLPath` or something may be more specific